### PR TITLE
feat: validate srcset ranges

### PR DESCRIPTION
**Pull request by ericdeansanchez**
The purpose of this PR is to validate "srcset ranges." A srcset
"range" is defined by it's `begin`, `end`, and `tol`.

If `tol` is less than `0.01` (one percent) then it is **invalid**.

If `begin > end`, then `begin` is **invalid**.

No restrictions have been explicitly applied to `end`; this is to
allow for flexibility when the maximum `end` is greater than 8192.

This PR also seeks to establish validation/error semantics for srcsets.
The question to be answered here in review is:

**Is this a sensible way of handling invalid input?**

Doing things this way means we do not have to throw exceptions. The
rationale is this:

- alert callers to an error state without affecting global process state

For instance, it seems likely that callers would rather deal with missing
images and empty srcset strings than with an exception that crashes
an app.

In short, an empty string means our code doesn't "pretend" things are
okay while failing gracefully within a larger application.

**Update**:
After some discussion, we have chosen to `throw` exceptions in these
exceptional cases. I am closing this PR and reopening another that
reflects this decision.

### CODE REVIEW

File: src/main/java/com/imgix/Validate.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+public class Validate {
+    /**
+     * Validates the width range specified by `begin` and `end`.
+     *
+     * A width range is valid if `begin` is less than or equal to
+     * `end` _and_ if `begin` is greater than zero.
+     *
+     * The maximum value for `end` is generally 8192. However, this
+     * is only the general case. In the future, values larger than
+     * 8192 will become valid.
```
Comment (sherwinski) [inline src/main/java/com/imgix/Validate.java:12]: Nit: I don't think we should be publicly promising this.
```suggestion
     * The maximum value for `end` is 8192, in
     * accordance with imgix's maximum renderable width.
     * However, we are not establishing a hard validation 
     * for it here, in the case that it ever changes.
```

File: src/main/java/com/imgix/Validate.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * Validate `widths` array to ensure it is _not_ empty.
+     *
+     * @param widths array of integer width values
+     * @return true if `widths` is non-empty, otherwise false
+     */
+    public static boolean isValidWidths(Integer[] widths) {
+        return widths.length > 0;
+    }
```
Comment (sherwinski) [inline src/main/java/com/imgix/Validate.java:52]: We should also validate that all the integers are positive, see [imgix-core-js](https://github.com/imgix/imgix-core-js/blob/master/src/imgix-core-js.js#L274-287).

**Review by sherwinski — CHANGES_REQUESTED** [review]
My 2¢, a strong reason in favor of throwing an exception is giving devs visibility on exactly _what went wrong_ and _where_. I do like the idea of gracefully failing as to opposed to halting at runtime, but I fear creating more friction for devs by placing the burden on them to figure out what went wrong and why.

File: src/main/java/com/imgix/Validate.java (hunk 1 above)
Comment (ericdeansanchez) [reply src/main/java/com/imgix/Validate.java:12]: I agree on the first part; have questions about this second part, but will remove this section nonetheless.

File: src/main/java/com/imgix/Validate.java (hunk 2 above)
Comment (ericdeansanchez) [reply src/main/java/com/imgix/Validate.java:52]: I strongly agree! Sorry, when I proposed we do this validation I was told we weren't going to, but I like that things have changed! I'll make the changes accordingly.

Comment (sherwinski) [reply src/main/java/com/imgix/Validate.java:52]: All good!

**Comment by ericdeansanchez** [comment]
After some discussion, we have chosen to `throw` exceptions in these
exceptional cases. I am closing this PR and reopening another that
reflects this decision.
